ipv-api/src/main/java/uk/gov/di/authentication/ipv/helpers/IPVCallbackHelper.java

```diff
@@ -0,0 +1,338 @@
+        this.dynamoIdentityService = dynamoIdentityService;
+        this.dynamoService = dynamoService;
+        this.objectMapper = objectMapper;
+        this.sessionService = sessionService;
+        this.sqsClient = sqsClient;
+    }
+
+    public APIGatewayProxyResponseEvent generateAuthenticationErrorResponse(
+            AuthenticationRequest authenticationRequest,
+            ErrorObject errorObject,
+            boolean noSessionErrorResponse,
+            String clientSessionId,
+            String sessionId) {
+        LOG.warn(
+                "Error in IPV AuthorisationResponse. ErrorCode: {}. ErrorDescription: {}. No Session Error: {}",
+                errorObject.getCode(),
+                errorObject.getDescription(),
+                noSessionErrorResponse);
+        auditService.submitAuditEvent(
+                IPVAuditableEvent.IPV_UNSUCCESSFUL_AUTHORISATION_RESPONSE_RECEIVED,
+                clientSessionId,
+                sessionId,
+                authenticationRequest.getClientID().getValue(),
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN);
+        var errorResponse =
+                new AuthenticationErrorResponse(
+                        authenticationRequest.getRedirectionURI(),
+                        errorObject,
+                        authenticationRequest.getState(),
+                        authenticationRequest.getResponseMode());
+        return generateApiGatewayProxyResponse(
+                302, "", Map.of(ResponseHeaders.LOCATION, errorResponse.toURI().toString()), null);
+    }
+
+    public AccountInterventionStatus getAccountInterventionStatus(
+            String internalPairwiseSubjectId) {
+        var accountInterventionStatus =
+                segmentedFunctionCall(
+                        "AIS: getAccountStatus",
+                        () ->
+                                this.accountInterventionService.getAccountStatus(
+                                        internalPairwiseSubjectId));
+        cloudwatchMetricsService.incrementCounter(
+                "AISResult",
+                Map.of(
+                        "blocked",
+                        String.valueOf(accountInterventionStatus.blocked()),
+                        "suspended",
+                        String.valueOf(accountInterventionStatus.suspended()),
+                        "resetPassword",
+                        String.valueOf(accountInterventionStatus.resetPassword()),
+                        "reproveIdentity",
+                        String.valueOf(accountInterventionStatus.reproveIdentity())));
+        return accountInterventionStatus;
+    }
+
+    public void doAccountIntervention(AccountInterventionStatus accountInterventionStatus) {
+        segmentedFunctionCall(
+                "AIS: getAccountStatus",
```

Comment:
This is the wrong segment name. I'm also not sure this needs to be a segmented function call. Are we specifically interested in the performance of this block?